extensions/mongodb-client/deployment/pom.xml

```diff
@@ -34,6 +34,12 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-mongodb-client</artifactId>
         </dependency>
+        <!-- TODO(SG): We may not need this - depending on whether we need to use the kubernetes-service-binding code in Processor or not -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-kubernetes-service-binding</artifactId>
+            <optional>true</optional>
+        </dependency>
```

Comment:
Best leave it in to make sure that Maven builds the modules in the correct order